consensus/scp/src/slot.rs

```diff
@@ -150,6 +150,92 @@ pub struct SlotMetrics {
     pub bN: u32,
 }
 
+/// Serializable slot state used for debugging purposes.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct SlotState<V: Value> {
```

Comment:
As mentioned in the PR description, this is not a particularly elegant solution. We might want to move into having an instance of this object inside `Slot`, but for the time being I opted for making this PR as small and least-obtrusive as I can.

Reply:
Would you be OK with putting this in a separate file (e.g. slot_state.rs or slot_debug.rs)? slot.rs is already very long and complicated.